python/ray/data/dataset.py

```diff
@@ -131,6 +131,7 @@
     import torch
     import torch.utils.data
     from tensorflow_metadata.proto.v0 import schema_pb2
+    from torch._prims_common import DeviceLikeType
```

Comment:
We'd want to avoid importing private APIs as it's not guaranteed to be stable across PyTorch releases.